### DISCUSSION TITLE
Setting Splitter panel size programmatically

### DISCUSSION THREAD

**Discussion by csakis** [primefaces/community#470] [PrimeVue]
Is there a way to set the Splitter panel sizes programmatically?  The `size` property is not reactive; it's only useful to set the initial sizes of panels. 
I'd like to be able to resize my panels via a function without rre-rendering the content of the panels?
Is this possible?

**Comment by Rakasch** [comment]
You can probably workaround that.
The panel sizes are only based on the setting of the css for flex-basis.
From the Splitter.vue component:
`this.nextPanelElement.style.flexBasis = 'calc(' + newNextPanelSize + '% - ' + (this.panels.length - 1) * this.gutterSize + 'px)';`
gutterSize is 4px by default, maybe depends on your theme.
So you can just update the style for each SplitterPanel yourself.

If you also want to use stateful, you will have to update this manually too.
Add a ref to splitter, update splitter.panelSizes and call splitter.saveState().

**Comment by saffireii** [comment]
I found that I could use a ref for the size, change the value programmatically and then call `resetState` and it would refresh the screen just like if I had dragged the bar manually.

```
<Button label="Swap Panel Sizes" @click="swapSizes" />
<Splitter ref="programmaticSplitter" @resizeend="manualResized">
  <SplitterPanel :size="panel1Size">contents</SplitterPanel>
  <SplitterPanel :size="panel2Size">bigger contents </SplitterPanel>
</Splitter>

<script setup>
import {ref, useTemplateRef} from 'vue'

const programmaticSplitter = useTemplateRef('programmaticSplitter')
const panel1Size = ref(20)
const panel2Size = ref(80)

const swapSizes = () => {
  const currentSizes = [panel1Size.value, panel2Size.value]

  panel1Size.value = currentSizes[1]
  panel2Size.value = currentSizes[0]

  programmaticSplitter.value.resetState()
}

const manualResized = () => {
  // handle anything you need to when the user drags to resize
  // in our case we are using Select buttons so I clear the ref that holds the selected button so that none of the buttons are "selected"
}

</script>
```
